pkg/backend/snapshot.go

```diff
@@ -123,7 +123,14 @@ func (sm *SnapshotManager) mutate(mutator func() bool) error {
 // Note that this is completely not thread-safe and defeats the purpose of having a `mutate` callback
 // entirely, but the hope is that this state of things will not be permament.
 func (sm *SnapshotManager) RegisterResourceOutputs(step deploy.Step) error {
-	return sm.mutate(func() bool { return true })
+	return sm.mutate(func() bool {
+		if old := step.Old(); old != nil && step.Old().Outputs.DeepEquals(step.New().Outputs) {
```

Comment:
Is there any chance that `step.New()` could be `nil`?

Reply:
There shouldn't be. If we have outputs, we _should_ have a non-`nil` new state. I am happy to add a `nil` check here for safety, though.

Reply:
Added an additional `nil` check